src/main/java/com/eternalcode/combat/fight/effect/EffectService.java

```diff
@@ -0,0 +1,87 @@
+package com.eternalcode.combat.fight.effect;
+
+import org.bukkit.entity.Player;
+import org.bukkit.potion.PotionEffect;
+import org.bukkit.potion.PotionEffectType;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+import java.util.List;
+import java.util.ArrayList;
+
+
+
+public class EffectService {
+    private final Map<UUID, List<PotionEffect>> activeEffects;
+
+    public EffectService() {
+        this.activeEffects = new HashMap<>();
+    }
+
+    public void storeActiveEffect(Player player, PotionEffect effect) {
+        List<PotionEffect> effects = this.activeEffects.computeIfAbsent(player.getUniqueId(), k -> new ArrayList<>());
+        effects.add(effect);
+
+    }
+
+    public void restoreActiveEffects(Player player) {
+        List<PotionEffect> effectsFromService = this.getCurrentEffects(player);
+
+        for (PotionEffect effect : effectsFromService) {
+            player.addPotionEffect(effect);
+        }
+
+        this.clearStoredEffects(player);
+    }
+
+    public void clearStoredEffects(Player player) {
+        List<PotionEffect> playerEffects = this.activeEffects.get(player.getUniqueId());
```

Comment:
You don't have to check if it exists in map - if not, it won't cause an error.
```suggestion
        this.activeEffects.remove(player.getUniqueId());
```